linen_examples/pixelcnn/train.py

```diff
@@ -30,97 +30,40 @@
 # See issue #620.
 # pytype: disable=wrong-keyword-args
 
-from functools import partial
+import functools
 import datetime
 
-from absl import app
-from absl import flags
 from absl import logging
-
-import numpy as np
-
 from flax import jax_utils
 from flax import optim
+import input_pipeline
+import pixelcnn
 from flax.metrics import tensorboard
 from flax.training import checkpoints
 from flax.training import common_utils
-
 import jax
-from jax import random
-from jax import lax
 import jax.numpy as jnp
-from jax.config import config
-
+import numpy as np
+import ml_collections
 import tensorflow as tf
 
-import input_pipeline
-import pixelcnn
-
-config.enable_omnistaging()
-
-FLAGS = flags.FLAGS
-
-flags.DEFINE_float(
-    'learning_rate', default=0.001, help=('The initial learning rate.'))
-
-flags.DEFINE_float(
-    'lr_decay',
-    default='0.999995',
-    help=('Learning rate decay, applied each optimization step.'))
-
-flags.DEFINE_integer(
-    'init_batch_size',
-    default=16,
-    help=('Batch size to use for data-dependent initialization.'))
 
-flags.DEFINE_integer(
-    'batch_size', default=64, help=('Batch size for training.'))
-
-flags.DEFINE_integer(
-    'num_epochs', default=200, help=('Number of training epochs.'))
-
-flags.DEFINE_float('dropout_rate', default=0.5, help=('DropOut rate.'))
-
-flags.DEFINE_integer(
-    'rng', default=0, help=('Random seed for network initialization.'))
-
-flags.DEFINE_integer(
-    'n_resnet', default=5, help=('Number of resnet layers per block.'))
-
-flags.DEFINE_integer(
-    'n_feature', default=160, help=('Number of features in each conv layer.'))
-
-flags.DEFINE_integer(
-    'n_logistic_mix',
-    default=10,
-    help=('Number of components in the output distribution.'))
-
-flags.DEFINE_string(
-    'model_dir',
-    default='./model_data',
-    help=('Directory to store model data.'))
-
-flags.DEFINE_float(
-    'polyak_decay',
-    default=0.9995,
-    help=('Exponential decay rate of the sum of previous model iterates '
-          'during Polyak averaging.'))
-
-
-def get_summary_writers():
+def get_summary_writers(workdir):
   current_time = datetime.datetime.now().strftime('%Y%m%d-%H%M%S')
-  log_dir = FLAGS.model_dir + '/log/' + current_time
+  log_dir = workdir + '/log/' + current_time
   train_log_dir = log_dir + '/train'
   eval_log_dir = log_dir + '/eval'
   train_summary_writer = tensorboard.SummaryWriter(train_log_dir)
   eval_summary_writer = tensorboard.SummaryWriter(eval_log_dir)
   return train_summary_writer, eval_summary_writer
 
 
-def model(**kwargs):
```

Comment:
This is another example of the pattern that @jheek has been rightfully criticizing (I admit I probably first introduced it out of haste! Sorry)

Do you think we'll be doing another pass on the examples changing this?

Reply:
Yes, I would be happy to pass around the initialized model instead of calling this function. But this PR is only big and I like to keep changes separate. This one introduced ConfigDict and the new file structure. Next will be a simple integration test and then I might change the model and summary writers.